cmd/kubeadm/app/cmd/token.go

```diff
@@ -130,24 +148,26 @@ func NewCmdTokenGenerate(out io.Writer) *cobra.Command {
 }
 
 // RunCreateToken generates a new bootstrap token and stores it as a secret on the server.
-func RunCreateToken(out io.Writer, cmd *cobra.Command, tokenDuration time.Duration, token string) error {
-	client, err := kubemaster.CreateClientFromFile(path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, kubeconfig.AdminKubeConfigFileName))
-	if err != nil {
-		return err
-	}
+func RunCreateToken(out io.Writer, client *clientset.Clientset, token string, tokenDuration time.Duration, usages []string) error {
 
-	parsedID, parsedSecret, err := tokenutil.ParseToken(token)
+	td := &kubeadmapi.TokenDiscovery{}
```

Comment:
If someone passes in a malformed token we should error explicitly.  If we instead ignore that and generate a random token they may think that everything worked until they try and used their malformed token elsewhere.

I'd rather fail early with clear message if the user passes a malformed token.